### PR TITLE
Avoid passing the LR Server's Channel Context to the Client Router.

### PR DESCRIPTION
**Pull request by pankti-m**
Description:
The Server's channel context is not required by the Client.  The code
can be decoupled and cleaned to remove this dependency.

Why should this be merged:  Necessary cleanup to allow role-based startup of the Client and Server in LR

Related issue(s) (if applicable): #<number>

## Checklist (Definition of Done):

- [ ] There are no TODOs left in the code
- [ ] [Coding conventions](https://github.com/CorfuDB/CorfuDB/wiki/Corfu-Style-Guidelines) (e.g. for logging, unit tests) have been followed
- [ ] Change is covered by automated tests
- [ ] Public API has Javadoc

### CODE REVIEW

**Comment by shama358** [comment]
Can we also remove it from IClientChannelAdapter?  https://github.com/CorfuDB/CorfuDB/blob/master/infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/transport/client/IClientChannelAdapter.java#L69

File: infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/transport/client/IClientChannelAdapter.java [hunk 1]
```diff
@@ -30,9 +30,6 @@ public abstract class IClientChannelAdapter {
     @Getter
     private final LogReplicationClientRouter router;
 
-    @Getter
-    private IChannelContext channelContext;
-
```
Comment (xcchang) [inline infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/transport/client/IClientChannelAdapter.java:35]: The import could be removed as well

**Review by xcchang — COMMENTED** [review]
I think not only for client side, the `IChannelContext` interface is never implemented or used ... so the interface itself could removed IMO

File: infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/transport/client/IClientChannelAdapter.java (hunk 1 above)
Comment (annym) [reply infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/transport/client/IClientChannelAdapter.java:35]: +1

Comment (annym) [reply infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/transport/client/IClientChannelAdapter.java:35]: +1, also we can remove it from IServerChannelAdapter and completely get rid of this interface like Chris suggested.

**Review by xcchang — APPROVED** [review]
LGTM!

**Review by shama358 — APPROVED** [review]

**Review by hisundar — APPROVED** [review]

**Review by Jas0nch — APPROVED** [review]

**Pull request merged**